src/js/game/LevelMVC/LevelModel.js

```diff
@@ -1175,30 +1175,30 @@ module.exports = class LevelModel {
           hasRight = true;
         }
 
-        if (y > 0 && !this.actionPlane._data[this.yToIndex(y - 1) + x].getIsEmptyOrEntity()) {
+        if (y > 0 && !this.actionPlane.getBlock(this.yToIndex(y - 1) + x).getIsEmptyOrEntity()) {
           // needs a bottom side AO shadow
           this.shadingPlane.push({ x: x, y: y, type: 'AOeffect_Bottom' });
         } else if (y > 0) {
-          if (x < this.planeWidth - 1 && !this.actionPlane._data[this.yToIndex(y - 1) + x + 1].getIsEmptyOrEntity() &&
-            this.actionPlane._data[this.yToIndex(y) + x + 1].getIsEmptyOrEntity()) {
+          if (x < this.planeWidth - 1 && !this.actionPlane.getBlock(this.yToIndex(y - 1) + x + 1).getIsEmptyOrEntity() &&
+            this.actionPlane.getBlock(this.yToIndex(y) + x + 1).getIsEmptyOrEntity()) {
             // needs a bottom left side AO shadow
             this.shadingPlane.push({ x: x, y: y, type: 'AOeffect_BottomLeft' });
           }
 
-          if (!hasRight && x > 0 && !this.actionPlane._data[this.yToIndex(y - 1) + x - 1].getIsEmptyOrEntity()) {
+          if (!hasRight && x > 0 && !this.actionPlane.getBlock(this.yToIndex(y - 1) + x - 1).getIsEmptyOrEntity()) {
             // needs a bottom right side AO shadow
             this.shadingPlane.push({ x: x, y: y, type: 'AOeffect_BottomRight' });
           }
         }
 
         if (y < this.planeHeight - 1) {
-          if (x < this.planeWidth - 1 && !this.actionPlane._data[this.yToIndex(y + 1) + x + 1].getIsEmptyOrEntity() &&
-            this.actionPlane._data[this.yToIndex(y) + x + 1].getIsEmptyOrEntity()) {
+          if (x < this.planeWidth - 1 && !this.actionPlane.getBlock(this.yToIndex(y + 1) + x + 1).getIsEmptyOrEntity() &&
+            this.actionPlane.getBlock(this.yToIndex(y) + x + 1).getIsEmptyOrEntity()) {
             // needs a bottom left side AO shadow
             this.shadingPlane.push({ x: x, y: y, type: 'AOeffect_TopLeft' });
           }
 
-          if (!hasRight && x > 0 && !this.actionPlane._data[this.yToIndex(y + 1) + x - 1].getIsEmptyOrEntity()) {
+          if (!hasRight && x > 0 && !this.actionPlane.getBlock(this.yToIndex(y + 1) + x - 1).getIsEmptyOrEntity()) {
```

Comment:
note that these could probably be updated to use `getBlockAt` with the offsets already provided by that method, but that's slightly out of scope for this PR

Reply:
I think we probably want to abstract away the underlying array that's storing the data. Consumers shouldn't know/care that the blocks are stored as a flat array vs an array of arrays vs some other data structure.

Is there anywhere we're accessing blocks by index that can't be written more cleanly with position?

Reply:
There are mostly a couple of places where we don't care and just want to be able to iterate through all the blocks in some order; there are also plenty of other places that are still using index internally for other purposes and which would require converting from index to position if we wanted to only support `getBlockAt` and never just `getBlock`.

I agree that I'd like to move more in the direction of just using position everywhere, but this feels like a good place for us to be at for now

Reply:
Alternatively, we could just add a method to get all the blocks for those iteration purposes and then update all the other places that are using index internally to convert to position, thus doing a better job of internally communicating that we really actually want to just always be using position.

The more I think about it the more I think I prefer that.

Reply:
Okay, yep, I like that _much_ better. `getBlock` is dead, long live `getBlockAt`